arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/Enviroment.kt

```diff
@@ -0,0 +1,86 @@
+package arrow.fx.coroutines
+
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.startCoroutine
+
+/**
+ * Environment that can run [suspend] programs using [startCoroutine] and [startCoroutineCancellable].
+ *
+ * An [Environment] runs on a certain [CoroutineContext] which is used to start the programs on.
+ * Since coroutines always return where they were started, this [CoroutineContext] also defines where you return after operators like [sleep] or [evalOn].
+ * Therefore it's advised to always run on [ComputationPool] which is the default setting.
+ *
+ * [Environment] also has an [asyncErrorHandler] which by default redirects to [Throwable.printStackTrace],
+ * no user flow errors will ever be send here [CancelToken] exceptions might bubble up here when they cannot be redirect to the user.
+ *
+ * This [Environment] is meant to be used in Java frameworks, or frameworks that do not expose [suspend] edge-points.
+ *
+ * This contract could be elaborated on Android to provide automatic cancellation on Android LifecycleOwner.
+ */
+interface Environment {
+
+  /**
+   * Start [CoroutineContext] of the programs ran using [startCoroutine] and [startCoroutineCancellable].
+   */
+  val ctx: CoroutineContext
+
+  /**
+   * The async error handler is reserved for rare exceptions.
+   * When an error occurs after the completion of a [kotlin.coroutines.Continuation],
+   * or when an error occurs when triggering a [CancelToken].
+   *
+   * This will never be invoked for a [Throwable] that occurred in a user flow.
+   */
+  fun asyncErrorHandler(e: Throwable): Unit
+
+  /**
+   * Execution strategy that will block the current thread that's waiting for the program to yield a value.
+   */
+  fun <A> unsafeRunSync(fa: suspend () -> A): A
+
+  /**
+   * Execution strategies that will immediately return and perform the program's work without blocking the current thread.
+   * This operation runs uncancellable.
```

Comment:
It may be worth adding a note about how exceptions are handled in this particular case.